compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeSceneMediator.uikit.kt

```diff
@@ -162,12 +199,61 @@ internal class ComposeSceneMediator(
         )
     }
 
-    private val delegate: SkikoUIViewDelegate by lazy {
-        SkikoUIViewDelegateImpl(
-            { scene },
-            interopContext,
-            densityProvider,
-        )
+    private val touchesDelegate: TouchesDelegate by lazy {
+        object : TouchesDelegate {
+            override fun pointInside(point: CValue<CGPoint>, event: UIEvent?): Boolean =
+                point.useContents {
+                    val position = this.toDpOffset().toOffset(densityProvider())
+                    !scene.hitTestInteropView(position)
+                }
+
+            override fun onTouchesEvent(view: UIView, event: UIEvent, phase: UITouchesEventPhase) {
+                scene.sendPointerEvent(
+                    eventType = phase.toPointerEventType(),
+                    pointers = event.touchesForView(view)?.map {
+                        val touch = it as UITouch
+                        val id = touch.hashCode().toLong()
+                        val position = touch.offsetInView(view, densityProvider().density)
+                        ComposeScenePointer(
+                            id = PointerId(id),
+                            position = position,
+                            pressed = touch.isPressed,
+                            type = PointerType.Touch,
+                            pressure = touch.force.toFloat(),
+                            historical = event.historicalChangesForTouch(
+                                touch,
+                                view,
+                                densityProvider().density
+                            )
+                        )
+                    } ?: emptyList(),
+                    timeMillis = (event.timestamp * 1e3).toLong(),
+                    nativeEvent = event
+                )
+            }
+        }
+    }
+
+    val renderDelegate by lazy {
```

Comment:
What do you think about renaming `RenderDelegate` back to `SkikoUIViewDelegate` or make it inner `Delegate` interface for `SkikoUIView`? It doesn't seem to be a general case interface whose functions are invoked by something else. Perhaps `SkikoUIView` could be renamed to `RenderingUIView`.

Reply:
Good suggestion, Done